lib/attr_comparable.rb

```diff
@@ -22,7 +22,9 @@ def compare_with_nil_code(left, right)
         elsif #{ right }.nil?
           1
         else
-          #{ left } <=> #{ right }
+          cmp_result = (#{ left } <=> #{ right })
+          return nil if cmp_result.nil?
```

Comment:
These two lines are related, but training `if` makes that hard to spot. Better to combine them:
```
if (cmp_result = (#{ left } <=> #{ right })).nil?
  return nil
end
```